src/Sentry.AspNetCore.Grpc/Extensions/DependencyInjection/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,40 @@
+using System.ComponentModel;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Sentry.AspNetCore;
+using Sentry.AspNetCore.Grpc;
+using Sentry.Extensibility;
+using Sentry.Extensions.Logging.Extensions.DependencyInjection;
+using Sentry.Extensions.Protobuf;
+
+// ReSharper disable once CheckNamespace -- Discoverability
+namespace Microsoft.Extensions.DependencyInjection
+{
+    /// <summary>
+    /// Extension methods for <see cref="IServiceCollection"/>
+    /// </summary>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    internal static class ServiceCollectionExtensions
+    {
+        /// <summary>
+        /// Adds Sentry's services to the <see cref="IServiceCollection"/>
+        /// </summary>
+        /// <param name="services">The services.</param>
+        /// <returns></returns>
+        public static IServiceCollection AddSentryGrpc(this IServiceCollection services)
+        {
+            _ = services.AddSingleton<ISentryEventProcessor, AspNetCoreEventProcessor>();
+            services.TryAddSingleton<IUserFactory, DefaultUserFactory>();
+
+            _ = services.AddSingleton<IProtobufRequestPayloadExtractor, DefaultProtobufRequestPayloadExtractor>();
```

Comment:
```suggestion
            _ = services.AddSingleton<IProtobufRequestPayloadExtractor, DefaultProtobufRequestPayloadExtractor>();
```

Looks like instead of copying the whole ASP.NET Core integration, all we need is this line ^, and the one below:

```
    services.AddGrpc(options =>
            {
                options.Interceptors.Add<SentryGrpcInterceptor>();
            });
```

Besides these, the code that goes in the middleware etc...

This integration could be be done differently then, we could offer a single entrypoint, and add it like:

`UseSentry(o => o.AddGrpc()`

And through that extension method we find the right hooks, like adding stuff into DI and the middleware.